examples/api/lib/widgets/scroll_view/listview_select.1.dart

```diff
@@ -0,0 +1,231 @@
+
+  @override
+  ListTileSelectExampleState createState() => ListTileSelectExampleState();
+}
+
+class ListTileSelectExampleState extends State<ListTileSelectExample> {
+  bool isEditMode = false;
+  final int listLength = 30;
+  late List<bool> _selected;
+  bool _selectAll = false;
+  bool _isGridMode = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _selected = List<bool>.generate(listLength, (_) => false);
+  }
+
+  @override
+  void dispose() {
+    _selected.clear();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text(
+            'ListTile select example',
+          ),
+          leading: isEditMode
+              ? IconButton(
+                  icon: const Icon(Icons.close),
+                  onPressed: () {
+                    setState(() {
+                      isEditMode = false;
+                    });
+                  },
+                )
+              : const BackButton(),
+          actions: <Widget>[
+            if (_isGridMode)
+              IconButton(
+                icon: const Icon(Icons.grid_on),
+                onPressed: () {
+                  setState(() {
+                    _isGridMode = false;
+                  });
+                },
+              )
+            else
+              IconButton(
+                icon: const Icon(Icons.list),
+                onPressed: () {
+                  setState(() {
+                    _isGridMode = true;
+                  });
+                },
+              ),
+            if (isEditMode)
+              TextButton(
+                  child: !_selectAll
+                      ? const Text('select all')
+                      : const Text('unselect all'),
+                  onPressed: () {
+                    _selectAll = !_selectAll;
+                    setState(() {
+                      _selected =
+                          List<bool>.generate(listLength, (_) => _selectAll);
+                    });
+                  }),
+          ],
+        ),
+        body: _isGridMode
+            ? GridBuilder(
+                isEditMode: isEditMode,
+                selectedList: _selected,
+                onEditChange: (bool x) {
+                  setState(() {
+                    isEditMode = x;
+                  });
+                },
+              )
+            : ListBuilder(
+                isEditMode: isEditMode,
+                selectedList: _selected,
+                onEditChange: (bool x) {
+                  setState(() {
+                    isEditMode = x;
+                  });
+                },
+              ));
+  }
+}
+
+// ignore: must_be_immutable
+class GridBuilder extends StatefulWidget {
+  GridBuilder(
+      {Key? key,
+      required this.selectedList,
+      required this.isEditMode,
+      this.onEditChange})
+      : super(key: key);
```

Comment:
Nit: Is this dart format? It looks a bit off.

Reply:
I guess yes, I am using vs code which uses the default formatter which formats the code on save.

Reply:
Probably this is resolved now